helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt

```diff
@@ -0,0 +1,187 @@
+) {
+    private val packageIdsFile by option(
+        "--package-ids-file",
+        help = "The list of package ids to put into the output analyzer result in plain text with one entry per line."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val ortFile by option(
+        "--ort-file", "-o",
+        help = "The ORT file to write the generated synthetic analyzer result to."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val configFile by option(
+        "--config",
+        help = "The path to the ORT configuration file that configures the scan results storage."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .default(ortConfigDirectory.resolve(ORT_CONFIG_FILENAME))
+
+    private val configArguments by option(
+        "-P",
+        help = "Override a key-value pair in the configuration file. For example: " +
+                "-P ort.scanner.storages.postgres.schema=testSchema"
+    ).associate()
+
+    override fun run() {
+        val ids = packageIdsFile.readLines().filterNot { it.isBlank() }.map { Identifier(it.trim()) }
+        val packages = openDatabaseConnection().use { getScannedPackages(it, ids) }.filterMaxByDate()
+        val ortResult = createAnalyzerResult(packages)
+
+        println("Writing analyzer result with ${packages.size} packages to '${ortFile.absolutePath}'.")
+        writeOrtResult(ortResult, ortFile)
+    }
+
+    private fun openDatabaseConnection(): Connection {
+        val storageConfig = OrtConfiguration.load(configArguments, configFile).scanner.storages
+            ?.get("postgres") as? PostgresStorageConfiguration
+            ?: throw IllegalArgumentException("postgresStorage not configured.")
+
+        return DatabaseUtils.createHikariDataSource(
+            config = storageConfig,
+            applicationNameSuffix = ORTH_NAME,
+            maxPoolSize = 1
+        ).connection
+    }
+}
+
+private data class ScannedPackage(
+    val id: Identifier,
+    val provenance: Provenance,
+    val scanTime: Instant
+)
+
+private fun getScannedPackages(connection: Connection, ids: Collection<Identifier>): List<ScannedPackage> {
+    val query = """
+        SELECT 
+            s.identifier as id,
+            s.scan_result->>'provenance' as provenance,
+            (s.scan_result->'summary'->>'start_time')::timestamp as start_time
+        FROM 
+            scan_results s 
+        WHERE 
+            s.identifier = ANY(?);
```

Comment:
Nit: If you add `ORDER BY start_time DESC` you could simplify `filterMaxByDate()`.

Reply:
I prefer the function as-is, e.g. without assuming any ordering.